index.d.cts

```diff
@@ -0,0 +1,193 @@
+import { Db } from 'mongodb';
+import { RedisClientType as RedisClient } from 'redis';
+
+/** Class representing a JoSk task runner (cron). */
+export class JoSk {
+  /**
+   * Create a JoSk instance
+   * @param {object} opts - configuration object
+   * @param {boolean} [opts.debug] - Enable debug logging
+   * @param {function} [opts.onError] - Informational hook, called instead of throwing exceptions, see readme for more details
+   * @param {boolean} [opts.autoClear] - Remove obsolete tasks (any tasks which are not found in the instance memory during runtime, but exists in the database)
+   * @param {number} [opts.zombieTime] - Time in milliseconds, after this period of time - task will be interpreted as "zombie". This parameter allows to rescue task from "zombie mode" in case when: `ready()` wasn't called, exception during runtime was thrown, or caused by bad logic
+   * @param {function} [opts.onExecuted] - Informational hook, called when task is finished, see readme for more details
+   * @param {number} [opts.minRevolvingDelay] - Minimum revolving delay — the minimum delay between tasks executions in milliseconds
+   * @param {number} [opts.maxRevolvingDelay] - Maximum revolving delay — the maximum delay between tasks executions in milliseconds
+   */
+  constructor(opts?: {
+    debug?: boolean;
+    onError?: Function;
+    autoClear?: boolean;
+    zombieTime?: number;
+    onExecuted?: Function;
+    minRevolvingDelay?: number;
+    maxRevolvingDelay?: number;
+  });
+  debug: boolean;
+  onError: boolean | Function;
+  autoClear: boolean;
+  zombieTime: number;
+  onExecuted: boolean | Function;
+  isDestroyed: boolean;
+  minRevolvingDelay: number;
+  maxRevolvingDelay: number;
+  nextRevolutionTimeout: number;
+  tasks: {};
+  _debug: (...args: any[]) => void;
+  adapter: any;
+  /**
+   * @async
+   * @memberOf JoSk
+   * @name ping
+   * @description Check package readiness and connection to Storage
+   * @returns {Promise<object>}
+   * @throws {mix}
+   */
+  ping(): Promise<object>;
+  /**
+   * @async
+   * @memberOf JoSk
+   * Create recurring task (loop)
+   * @name setInterval
+   * @param {function} func - Function (task) to execute
+   * @param {number} delay - Delay between task execution in milliseconds
+   * @param {string} uid - Unique function (task) identification as a string
+   * @returns {Promise<string>} - Timer ID
+   */
+  setInterval(func: Function, delay: number, uid: string): Promise<string>;
```

Comment:
I see some improvement can be done.
For example the cronjob function, task structure that added to database, etc

Reply:
it can be strings like in case of Redis, or what do you mean exactly?

Reply:
For what I means is that the `Function` can be more specific in TypeScript.

The `task: object` can also be standardize to a proper struct.

Reply:
@climba03003 `function<any|Promise<any>>`?

Reply:
It would be something like
```
(ready: ((next: Date) => void) | ((callback: (err: Error, success: boolean) => void) => void)) => Promise<void> | void
```

Reply:
@climba03003 
1. Callback is gone
2. Should include only `ready`

Reply:
@climba03003 lmk if it's good to go or we still want to change something

Reply:
> lmk if it's good to go or we still want to change something

The current one is good to go.
I can fine tune it afterward.